src/v/kafka/server/handlers/produce.cc

```diff
@@ -302,7 +373,7 @@ produce_topic(produce_ctx& octx, produce_request::topic& topic) {
         }
 
         if (unlikely(!part.records->adapter.valid_crc)) {
-            partitions.push_back(
+            partitions_produced.push_back(
```

Comment:
why don't we need to also push something onto the `partitions_dispatched` vector like ss::now or something?